DMCompiler/Optimizer/PeepholeOptimizations.cs

```diff
@@ -458,6 +459,10 @@ public void Apply(DMCompiler compiler, List<IAnnotatedBytecode> input, int index
 
         IOptimization.GetInstructionAndValue(input[index + 1], out var pushVal2);
 
+        if (pushVal2 == 0) {
+            compiler.Emit(WarningCode.BadExpression, input[index + 1].GetLocation(), "Division by zero");
+        }
```

Comment:
This fixes `div_zero4.dm`

It would fix the other broken tests if we actually did const folding on them.

Reply:
We want peephole optimizations to be disable-able with CLI arguments someday right? Tying compile errors to them probably isn't a good idea.